lasagne/layers/recurrent.py

```diff
@@ -294,16 +311,16 @@ def __init__(self, incoming, input_to_hidden, hidden_to_hidden,
         # Check that input_to_hidden's output shape is the same as
         # hidden_to_hidden's input shape but don't check a dimension if it's
         # None for either shape
+        h_to_h_input_shape = hidden_to_hidden_in_layer.output_shape
         if not all(s1 is None or s2 is None or s1 == s2
                    for s1, s2 in zip(input_to_hidden.output_shape[1:],
-                                     hidden_to_hidden.input_shape[1:])):
-            raise ValueError("The output shape for input_to_hidden "
-                             "must be equal to the input shape of "
-                             "hidden_to_hidden after the first dimension, but "
-                             "input_to_hidden.output_shape={} and "
-                             "hidden_to_hidden.input_shape={}".format(
-                                 input_to_hidden.output_shape,
-                                 hidden_to_hidden.input_shape))
+                                     h_to_h_input_shape[1:])):
+            raise ValueError(
+                "The output shape for input_to_hidden must be equal to the "
+                "input shape of hidden_to_hidden after the first dimension, "
+                "but input_to_hidden.output_shape={} and "
+                "hidden_to_hidden:input_layer.output_shape={}".format(
```

Comment:
Maybe `hidden_to_hidden:input_layer.shape` is easier to grasp? I don't know. `InputLayer` has a `shape` property that may feel more natural than `output_shape`.